src/logic/new.ts

```diff
@@ -0,0 +1,166 @@
+import type { ProviderJson } from '@superfaceai/ast';
+import { parseProfile, Source } from '@superfaceai/parser';
+import type { ServiceClient } from '@superfaceai/service-client';
+
+import type { ILogger } from '../common';
+import { SuperfaceClient } from '../common/http';
+import { pollUrl } from '../common/polling';
+
+export type ProfilePreparationResponse = {
+  // Id of the profile with . separated scope and name
+  id: string;
+  // TODO: get AST from server to avoid parsing (possible problems with AST/Parser versioning)?
+  profile: {
+    source: string;
+  };
+};
+
+function assertProfileResponse(
+  input: unknown
+): asserts input is ProfilePreparationResponse {
+  if (
+    typeof input === 'object' &&
+    input !== null &&
+    'id' in input &&
+    'profile' in input
+  ) {
+    const tmp = input as { id: string; profile: { source?: string } };
+
+    if (typeof tmp.profile.source !== 'string') {
+      throw Error(
+        `Unexpected response received - missing profile source: ${JSON.stringify(
+          tmp,
+          null,
+          2
+        )}`
+      );
+    }
+
+    try {
+      parseProfile(new Source(tmp.profile.source));
+    } catch (e) {
+      throw Error(
+        `Unexpected response received - unable to parse profile source: ${JSON.stringify(
+          e,
+          null,
+          2
+        )}`
+      );
+    }
+
+    // TODO: validate id format?
```

Comment:
Parser validates the profile name. What kind of validation do you mean here?

Reply:
I was thinking about `.` or `/` separators of scope and name.  But probably it is not needed now.